src/App.jsx

```diff
@@ -2,18 +2,26 @@
 import { useRef, useState } from 'react';
 
 function App() {
+  const [isEditableList, setIsEditableList] = useState([]);
   const [todos, setTodos] = useState([]);
   const inputRef = useRef(null);
   const handleAddTodoClick = () => {
     const newTodo = inputRef.current.value;
     setTodos(todos.concat(newTodo));
     inputRef.current.value = '';
+    setIsEditableList(isEditableList.concat(false));
   };
   const handleDeleteClick = (index) => {
-    const newTodos = todos.filter((_, todoIndex) => todoIndex !== index);
+    const newTodos = [...todos.slice(0, index), ...todos.slice(index + 1)];
```

Comment:
you really achieved great progress there. some parts were missing but this is just a beginning